src/dataloader.ts

```diff
@@ -80,31 +82,35 @@ export class LoaderResolver {
     }
     for (const [ext, Extractor] of extractors) {
       const archive = dir + ext;
-      if (existsSync(join(this.root, archive))) {
-        return new Extractor({
-          preload: async () => archive,
-          inflatePath: targetPath.slice(archive.length - ext.length + 1),
-          path: join(this.root, archive),
-          root: this.root,
-          targetPath,
-          useStale
-        });
-      }
-      const archiveLoader = this.findExact(archive, {useStale});
+      const archiveLoader = this.findExact(archive, {useStale}) ?? this.findDynamic(archive, {useStale});
```

Comment:
I think it would be more correct to compute the Cartesian cross-product of archive extensions (.zip, .tar, etc.) and interpreter extensions (.js, .py, etc.) here so that a more-specific parameterized match takes precedence over a less-specific parameterized match with a different archive extension. For example, you’d want /foo/bar.tar to take precedence over /[param]/bar.zip.

Although I think this is actually already broken because crawling the parent folders currently requires an exact match in order to look for an archive/extractor; so parameterized archives probably don’t work as intended. We need use glob to match `[param]` directory names. This will require further investigation. There are a lot of combinations of dynamic routing to consider! 😓